### PR TITLE
raftstore: fix the region meta inconsistency bug during splitting

### PR DESCRIPTION
**Pull request by gengliqi**
### What problem does this PR solve?

Issue Number: close https://github.com/tikv/tikv/issues/8783

Problem Summary:
https://github.com/tikv/tikv/issues/8783#issuecomment-708298415
The bug is located in function `on_ready_split_region` which is introduced by https://github.com/tikv/tikv/pull/8084.
https://github.com/tikv/tikv/blob/7c9850040ba58d48f0f1a93345c0392dffb2fc5b/components/raftstore/src/store/fsm/peer.rs#L2084-L2112
Line 2111 `continue` means this region has already existed so it should be skipped.
But line 2087-2090 has already added this region to `region_ranges` which leads to metadata inconsistency.

### What is changed and how it works?

What's Changed:
Fix this bug and add 2 tests.

Run `test_split_not_to_split_existed_tombstone_region ` and `test_split_not_to_split_existed_different_uninitialied_peer` can panic in different place in previous code.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test

Side effects

- No

### Release note 
* No release note

### CODE REVIEW

File: tests/failpoints/cases/test_split_region.rs [hunk 1]
```diff
@@ -248,7 +248,7 @@ impl Filter for PrevoteRangeFilter {
 // region id has already existed. In previous implementation, it can be created and panic
 // will happen because there are two initialized peer with the same region id.
 #[test]
-fn test_split_not_to_split_exist_region() {
+fn test_split_not_to_split_existed_region() {
```
Comment (BusyJay) [inline tests/failpoints/cases/test_split_region.rs:251]: s/existed/existing/

**Review by BusyJay — COMMENTED** [review]
Rest LGTM

File: tests/failpoints/cases/test_split_region.rs [hunk 2]
```diff
@@ -378,8 +378,126 @@ fn test_split_not_to_split_exist_tombstone_region() {
...
+    cluster.must_split(&region, b"k2");
+    let left = pd_client.get_region(b"k1").unwrap();
+    assert_eq!(left.get_id(), 1000);
+    let left_peer_2 = find_peer(&left, 2).cloned().unwrap();
+
+    pd_client.must_remove_peer(left.get_id(), left_peer_2);
+    pd_client.must_add_peer(left.get_id(), new_peer(2, 4));
+
+    let before_check_snapshot_1000_2_fp = "before_check_snapshot_1000_2";
+    fail::cfg(before_check_snapshot_1000_2_fp, "pause").unwrap();
```
Comment (NingLin-P) [inline tests/failpoints/cases/test_split_region.rs:485]: How about setting the fail point before splitting?

**Review by NingLin-P — COMMENTED** [review]
Rest LGTM

File: tests/failpoints/cases/test_split_region.rs (hunk 2 above)
Comment (gengliqi) [reply tests/failpoints/cases/test_split_region.rs:485]: There is no difference because there is a send filter of store 2.

**Review by NingLin-P — APPROVED** [review]

**Review by BusyJay — APPROVED** [review]

**Pull request merged**